MinnPost/object-sync-for-salesforce
# Allow fieldmaps to choose a default status for new records in WordPress

## What does this PR do?

This fixes #319. It adds an Ajax (and server side, once the fieldmap has a WordPress object type) field to the fieldmap create/edit screen. With it, users can set a status for the record WordPress will create, if WordPress supports statuses on that object (posts and comments do).

If the fieldmap already maps a status field, or if the user is using the developer hook, it should override this value when modifying data.

## How do I test this PR?

- map an object that supports statuses
- make sure the status field gets populated by the default value
- make sure mapping a field to the status, or using the developer hook to set the status, overrides the default value
